app/Services/Subusers/SubuserUpdateService.php

```diff
@@ -16,25 +17,26 @@ public function __construct(
 
     public function handle(Subuser $subuser, Server $server, array $permissions): void
     {
+        $cleanedPermissions = array_filter($permissions, fn ($permission) => $permission === Permission::ACTION_WEBSOCKET_CONNECT || auth()->user()->can($permission, $server));
```

Comment:
```suggestion
```suggestion
            $cleanedPermissions = collect($permissions)
                ->unique()
                ->filter(fn ($permission) => auth()->user()->can($permission, $server))
                ->sort()
                ->values()
                ->all();
```
Otherwise we get `{"4":"user.read","5:"user.create","6:"user.update,"7:"user.delete"` instead of `["user.create","user.delete","user.read","user.update"]`

Also why should a user that can't connect to the WEBSOCKET be able to grant that ?
```